pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -665,6 +719,8 @@ func (s *Reconciler) createVirtualMachine(ctx context.Context, nicName string) e
 			return errors.Errorf("vm %s is nil provisioning state, reconcile", s.scope.Machine.Name)
 		}
 
+		s.setPrettyMachineAnnotations(vm)
```

Comment:
coudn't we call this in the upper level methods and get anything needed from `scope.MachineConfig` and remove the logic around `ProvisioningState`?

Reply:
No, we could not. Annotations are extracted from vm object which reflects current state of vm as the Azure cloud providers sees it. Reading the values from `scope.MachineConfig` would not reflect the current state.

Reply:
can we do:
- labels for region/zone/type
- instanceID is already available in `machine.Spec.ProviderID` so nothing to do
- state: dumb annotation copying whatever there's in `scope.MachineStatus.VMState` / or enabling `machine.Status.State` See https://github.com/openshift/cluster-api-provider-azure/pull/59#discussion_r306194991
This is purely informative, similar to https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#built-in-node-labels

Reply:
Good point with using labels for region (`failure-domain.beta.kubernetes.io/region`) and zone (`failure-domain.beta.kubernetes.io/zone`) (users will be able to list all machines for a given region/zone). Instance type label (`beta.kubernetes.io/instance-type`) for filtering e.g. gpu machines is useful as well.

In case of `instanceID`, it's definitely useful for AWS, where you can search for an instance just using the instance ID. The same does not hold for Azure so the field might be just left empty. I was not able to find even any CLI command using vmid for a look up.

> state: dumb annotation copying whatever there's in scope.MachineStatus.VMState

That's what I already do. Just reading the state directly from the vm object. Also, `VMState` has nothing for a running state so my implementation actually supersedes what we store under `scope.MachineStatus.VMState`.